### PR TITLE
tests: skip boot state test on arm devices

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Skip on ubuntu-core-(18|20)-arm-* due to fw_printenv/setenv are not provided by the systems

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you

File: tests/main/boot-state/task.yaml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 summary: smoke test for the boot-state tool
 
+# ARM devices are not supported on ubuntu-core-18/20 due to fw_printenv/setenv are
+# not provided by the system and as the devices boot with uboot so it is not
+# possible to get any grub information as it is done with non arm devices.
+systems: [-ubuntu-core-18-arm-*, -ubuntu-core-20-arm-*]
```
Comment (anonymouse64) [inline tests/main/boot-state/task.yaml:6]: if we don't execute this test on uc20 arm, why do we have this check below in the test:


```bash
    case "$SPREAD_SYSTEM" in
        ubuntu-core-16-arm-*|ubuntu-core-18-arm-*|ubuntu-core-20-arm-*)
            "$TESTSTOOLS"/boot-state boot-path | MATCH "/boot/uboot/"
            ;;
```
on line 33-34

either we should drop those cases from the test, or we should refactor this test to only do that part of the test on uc20 ARM I think

File: tests/main/boot-state/task.yaml [hunk 2]
```diff
@@ -1,5 +1,10 @@
 summary: smoke test for the boot-state tool
 
+# ARM devices are not supported on ubuntu-core-18/20 due to fw_printenv/setenv are
```
Comment (anonymouse64) [inline tests/main/boot-state/task.yaml:3]: perhaps we should eventually try to support doing this kind of test not with `fw_printenv`/`setenv` and friends but instead through `snap debug boot-state ...` much like we do with `snap debug boot-vars`. If that's the case I would suggest we leave a TODO: to that effect in this file or in boot-state to suggest that refactoring (I didn't see anything about that refactoring on master).

**Review by anonymouse64 — COMMENTED** [review]
a couple comments

File: tests/main/boot-state/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/main/boot-state/task.yaml]: thanks for the suggestions, I already updated the test to run in arm devices the part of the test which is possible to validate and then exit.

File: tests/main/boot-state/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/main/boot-state/task.yaml]: done

**Review by anonymouse64 — APPROVED** [review]
LGTM thanks

**Pull request merged**